test/middleware_test.rb

```diff
@@ -27,65 +30,74 @@ def call(env)
           middleware.call(env)
-          expect(app.env).to include(env)
+          _(app.env).must_equal(env)
         end
 
         it "starts the reporter" do
           middleware.call(env)
-          expect(Reporter.instance).to be_started
+          _(Reporter.instance).must_be :started?
         end
 
-        context "when the request includes HTTP_X_REQUEST_START" do
+        describe "when the request includes HTTP_X_REQUEST_START" do
           let(:five_seconds_ago_in_unix_millis) { (Time.now.to_f - 5) * 1000 }
+
           before { env["HTTP_X_REQUEST_START"] = five_seconds_ago_in_unix_millis.to_i.to_s }
-          before { Singleton.__init__(Store) }
+          after { Store.instance.clear }
 
           it "collects the request queue time" do
             middleware.call(env)
 
             report = Store.instance.pop_report
-            expect(report.measurements.length).to eql 1
-            expect(report.measurements.first).to be_a Measurement
-            expect(report.measurements.first.value).to be_within(1).of(5000)
+            _(report.measurements.length).must_equal 1
+            _(report.measurements.first).must_be_instance_of Measurement
+            _(report.measurements.first.value).must_be_within_delta 5000, 1
           end
 
           it "records the queue time in the environment passed on" do
             middleware.call(env)
 
-            expect(app.env).to have_key("queue_time")
-            expect(app.env["queue_time"]).to be_within(1).of(5000)
+            _(app.env).must_include("queue_time")
+            _(app.env["queue_time"]).must_be_within_delta 5000, 1
           end
 
-          context "when the request body is large enough to skew the queue time" do
+          describe "when the request body is large enough to skew the queue time" do
             before { env["rack.input"] = StringIO.new("." * 110_000) }
 
             it "does not collect the request queue time" do
               middleware.call(env)
 
               report = Store.instance.pop_report
-              expect(report.measurements.length).to eql 0
+              _(report.measurements.length).must_equal 0
             end
           end
         end
       end
 
-      context "without JUDOSCALE_URL set" do
-        around { |example| use_env({"JUDOSCALE_URL" => nil}, &example) }
+      describe "without JUDOSCALE_URL set" do
+        def setup
+          use_env({"JUDOSCALE_URL" => nil}) do
+            super
+          end
+        end
 
         it "passes the request up the middleware stack" do
           middleware.call(env)
-          expect(app.env).to include(env)
+          _(app.env).must_equal env
         end
 
         it "does not start the reporter" do
-          allow(Reporter.instance).to receive(:register!)
-          middleware.call(env)
-          expect(Reporter.instance).to_not have_received(:register!)
+          Reporter.instance.stub(:register!, -> { raise "NEVER CALLED" }) do
```

Comment:
```suggestion
          Reporter.instance.stub(:register!, -> { raise "SHOULD NOT BE CALLED" }) do
```

Just a slight clarification here since I think it might be confusing to see a "never called" exception when a method *was* called.